File: elevaite/python_apps/agent_studio/agent-studio/db/schemas/workflows.py

import uuid
from datetime import datetime
from typing import Dict, List, Optional, Literal, Any

from pydantic import BaseModel, ConfigDict

from .agents import AgentResponse

class WorkflowBase(BaseModel):
    name: str
    description: Optional[str] = None
    version: str = "1.0.0"
    configuration: Dict[str, Any]
    created_by: Optional[str] = None
    is_active: bool = True
    tags: Optional[List[str]] = None

class WorkflowCreate(WorkflowBase):
    pass

class WorkflowUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    version: Optional[str] = None
    configuration: Optional[Dict[str, Any]] = None
    is_active: Optional[bool] = None
    tags: Optional[List[str]] = None

class WorkflowInDB(WorkflowBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    workflow_id: uuid.UUID
    created_at: datetime
    updated_at: datetime
    is_deployed: bool
    deployed_at: Optional[datetime] = None

class WorkflowResponse(WorkflowInDB):
    workflow_agents: List["WorkflowAgentResponse"] = []
    workflow_connections: List["WorkflowConnectionResponse"] = []
    workflow_deployments: List["WorkflowDeploymentResponse"] = []

class WorkflowAgentBase(BaseModel):
    workflow_id: uuid.UUID
    agent_id: uuid.UUID
    position_x: Optional[int] = None
    position_y: Optional[int] = None
    node_id: str
    agent_config: Optional[Dict[str, Any]] = None

class WorkflowAgentCreate(WorkflowAgentBase):
    pass

class WorkflowAgentUpdate(BaseModel):
    position_x: Optional[int] = None
    position_y: Optional[int] = None
    node_id: Optional[str] = None
    agent_config: Optional[Dict[str, Any]] = None

class WorkflowAgentInDB(WorkflowAgentBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    added_at: datetime

class WorkflowAgentResponse(WorkflowAgentInDB):
    agent: AgentResponse

class WorkflowConnectionBase(BaseModel):
    workflow_id: uuid.UUID
    source_agent_id: uuid.UUID
    target_agent_id: uuid.UUID
    connection_type: str = "default"
    conditions: Optional[Dict[str, Any]] = None
    priority: int = 0
    source_handle: Optional[str] = None
    target_handle: Optional[str] = None

class WorkflowConnectionCreate(WorkflowConnectionBase):
    pass

class WorkflowConnectionUpdate(BaseModel):
    connection_type: Optional[str] = None
    conditions: Optional[Dict[str, Any]] = None
    priority: Optional[int] = None
    source_handle: Optional[str] = None
    target_handle: Optional[str] = None

class WorkflowConnectionInDB(WorkflowConnectionBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    created_at: datetime

class WorkflowConnectionResponse(WorkflowConnectionInDB):
    source_agent: AgentResponse
    target_agent: AgentResponse

class WorkflowDeploymentBase(BaseModel):
    workflow_id: uuid.UUID
    environment: str = "production"
    deployment_name: str
    deployed_by: Optional[str] = None
    runtime_config: Optional[Dict[str, Any]] = None

class WorkflowDeploymentCreate(WorkflowDeploymentBase):
    pass

class WorkflowDeploymentRequest(BaseModel):

    environment: str = "production"
    deployment_name: str
    deployed_by: Optional[str] = None
    runtime_config: Optional[Dict[str, Any]] = None

class WorkflowDeploymentUpdate(BaseModel):
    status: Optional[Literal["active", "inactive", "failed"]] = None
    runtime_config: Optional[Dict[str, Any]] = None
    last_error: Optional[str] = None

class WorkflowDeploymentInDB(WorkflowDeploymentBase):
    id: int
    deployment_id: uuid.UUID
    status: Literal["active", "inactive", "failed"] = "active"
    deployed_at: datetime
    stopped_at: Optional[datetime] = None
    last_executed: Optional[datetime] = None
    execution_count: int = 0
    error_count: int = 0
    last_error: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)

class WorkflowDeploymentResponse(WorkflowDeploymentInDB):
    workflow: WorkflowInDB

class WorkflowExecutionRequest(BaseModel):
    workflow_id: Optional[uuid.UUID] = None
    deployment_name: Optional[str] = None
    query: str
    chat_history: Optional[List[Dict[str, str]]] = None
    runtime_overrides: Optional[Dict[str, Any]] = None

class WorkflowStreamExecutionRequest(BaseModel):
    deployment_name: str
    query: str
    chat_history: Optional[List[Dict[str, str]]] = None
    runtime_overrides: Optional[Dict[str, Any]] = None

class WorkflowExecutionResponse(BaseModel):
    status: str
    response: Optional[str] = None
    execution_id: Optional[str] = None
    workflow_id: Optional[str] = None
    timestamp: str

File: elevaite/apps/command_agent_version1/app/lib/interfaces/workflows.ts
import { AgentResponse } from "./agents";

export interface WorkflowPosition {
  x: number;
  y: number;
}

export interface WorkflowAgentData {
  agent_id: string;
  position?: WorkflowPosition;
}

export interface WorkflowConnectionData {
  source_agent_id: string;
  target_agent_id: string;
  connection_type?: "default" | "conditional" | "parallel";
}

export interface WorkflowDeployRequest {
  workflow_name?: string;
  description?: string;
  agents: WorkflowAgentData[];
  connections: WorkflowConnectionData[];
}

export interface WorkflowDeployResponse {
  status: "success" | "error";
  message: string;
  workflow_id?: string;
  deployment_id?: string;
  deployment_name?: string;
}

export interface WorkflowExecutionRequest {
  query: string;
  uid?: string;
  sid?: string;
  collection?: string;
}

export interface WorkflowExecutionResponse {
  status: "ok" | "error";
  response?: string;
  message?: string;
}

export interface WorkflowAgent {
  workflow_id: string;
  agent_id: string;
  position_x?: number;
  position_y?: number;
  node_id: string;
  agent_config?: Record<string, any>;
  id: number;
  added_at: string;
  agent: AgentResponse;
}

export interface WorkflowConnection {
  workflow_id: string;
  source_agent_id: string;
  target_agent_id: string;
  connection_type: string;
  conditions?: Record<string, any>;
  priority: number;
  source_handle?: string;
  target_handle?: string;
  id: number;
  created_at: string;
  source_agent: AgentResponse;
  target_agent: AgentResponse;
}

export interface WorkflowDeployment {
  workflow_id: string;
  environment: string;
  deployment_name: string;
  deployed_by?: string;
  runtime_config?: Record<string, any>;
  id: number;
  deployment_id: string;
  status: "active" | "inactive" | "failed";
  deployed_at: string;
  stopped_at?: string;
  execution_count: number;
  error_count: number;
  last_executed?: string;
  last_error?: string;
  workflow: WorkflowResponse;
}

export interface WorkflowResponse {
  name: string;
  description?: string;
  version: string;
  configuration: Record<string, any>;
  created_by?: string;
  is_active: boolean;
  tags?: string[];
  id: number;
  workflow_id: string;
  created_at: string;
  updated_at: string;
  is_deployed: boolean;
  deployed_at?: string;
  workflow_agents: WorkflowAgent[];
  workflow_connections: WorkflowConnection[];
  workflow_deployments: WorkflowDeployment[];
}

export interface WorkflowCreateRequest {
  name: string;
  description?: string;
  version?: string;
  configuration: {
    agents: Array<{
      agent_type: string;
      agent_id?: string;
      position?: { x: number; y: number };
      config?: Record<string, any>;
    }>;
    connections: Array<{
      source_agent_id: string;
      target_agent_id: string;
      connection_type?: string;
      conditions?: Record<string, any>;
      priority?: number;
      source_handle?: string;
      target_handle?: string;
    }>;
  };
  created_by?: string;
  is_active?: boolean;
  tags?: string[];
}

export interface WorkflowDeploymentRequest {
  environment?: string;
  deployment_name: string;
  deployed_by?: string;
  runtime_config?: Record<string, any>;
}

export interface NewWorkflowExecutionRequest {
  workflow_id?: string;
  deployment_name?: string;
  query: string;
  chat_history?: Array<{ actor: string; content: string }>;
  runtime_overrides?: Record<string, any>;
}

export interface SavedWorkflow {
  workflow_id: string;
  name: string;
  description?: string;
  created_at: string;
  created_by?: string;
  is_active: boolean;
  is_deployed: boolean;
  deployed_at?: string;
  version: string;
  agent_count?: number;
  connection_count?: number;
}
